opensearch-operator/pkg/reconcilers/securityconfig_test.go

```diff
@@ -152,8 +239,55 @@ var _ = Describe("Securityconfig Reconciler", func() {
 			Expect(err).ToNot(HaveOccurred())
 
 			job := batchv1.Job{}
+			// Should throw an error as the update job should not exist
 			Expect(k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-securityconfig-update", Namespace: clusterName}, &job)).To(HaveOccurred())
 
 		})
 	})
+
+	When("When Reconciling the securityconfig reconciler with no securityconfig secret but tls configured", func() {
+		It("should start an update job and apply all yml files", func() {
+			var clusterName = "no-securityconfig-tls-configured"
+			// Create namespace and secret first
+			Expect(CreateNamespace(k8sClient, clusterName)).Should(Succeed())
+			spec := opsterv1.OpenSearchCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName, Namespace: clusterName, UID: "dummyuid"},
+				Spec: opsterv1.ClusterSpec{
+					General: opsterv1.GeneralConfig{
+						ServiceName: clusterName,
+						Version:     "2.3",
+					},
+					Security: &opsterv1.Security{
+						Tls: &opsterv1.TlsConfig{
+							Transport: &opsterv1.TlsConfigTransport{Generate: true},
+						},
+					},
+				}}
+
+			reconcilerContext := NewReconcilerContext(spec.Spec.NodePools)
+			underTest := NewSecurityconfigReconciler(
+				k8sClient,
+				context.Background(),
+				&helpers.MockEventRecorder{},
+				&reconcilerContext,
+				&spec,
+			)
+			_, err := underTest.Reconcile()
+			Expect(err).ToNot(HaveOccurred())
+
+			job := batchv1.Job{}
+			// Should not throw an error as the update job should exist
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-securityconfig-update", Namespace: clusterName}, &job)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			clusterHostName := BuildHostName(&spec)
+			httpPort, securityconfigPath := helpers.VersionCheck(&spec)
+			cmdArg := fmt.Sprintf(SecurityAdminBaseCmdTmpl, clusterHostName, httpPort) +
```

Comment:
I'd prefer you not use the constants here, it's not really god isolation of the test from the impl. Instead either construct the expected string by hand or test for some parts of the cmd string.

Reply:
Got it! Changed the code to construct the expected string by hand